test/test_integration.py

```diff
@@ -1230,6 +1231,114 @@ def do_connect(conn):
 
         self.tear_down()
 
+    def test_connection_timeout(self):
+        self.set_up(timeout=0.5)
+
+        def socket_handler(listener):
+            time.sleep(1)
+            sock = listener.accept()[0]
```

Comment:
No point calling `accept`: it'll usually fail, and throw exceptions, which we don't want.